Tests/SentryTests/SentryKSCrashReportConverterTests.m

```diff
@@ -69,6 +69,34 @@ - (void)testConvertReport
     XCTAssertNotNil([[event serialize] valueForKeyPath:@"threads.values"]);
 }
 
+/**
+ * Reproduces an issue for parsing a recrash report of a customer that leads to a crash.
+ * The report contains a string instead of a thread dictionary in crash -> threads.
+ * SentryCrashReportConverter expects threads to be a dictionary that contains the details about a
+ * thread.
+ */
+- (void)testRecrashReport_WithThreadIsStringInsteadOfDict
+{
+    NSDictionary *report = [self getCrashReport:@"Resources/recrash-report"];
+
+    SentryCrashReportConverter *reportConverter =
+        [[SentryCrashReportConverter alloc] initWithReport:report];
+    SentryEvent *event = [reportConverter convertReportToEvent];
+
+    // Do only a few basic assertions here. RecrashReport is tested with testUnknownTypeException
+    XCTAssertEqual(1, event.exceptions.count);
+    SentryException *exception = event.exceptions.firstObject;
+    XCTAssertEqualObjects(@"EXC_BAD_ACCESS", exception.type);
+    XCTAssertEqualObjects(@"Exception 1, Code 3657279596, Subcode 8", exception.value);
+
+    XCTAssertEqual(1, event.threads.count);
+    SentryThread *thread = event.threads.firstObject;
+    XCTAssertFalse(thread.crashed);
+    XCTAssertFalse(thread.current);
+    XCTAssertEqual(21, thread.stacktrace.frames.count);
+    XCTAssertEqual(21, thread.stacktrace.registers.count);
```

Comment:
M: This this test case is targeting `threads` in this String use case, we could get around removing this assert and deleting all those registers from the sample JSON (which is the largest part of this charge)
```suggestion
```

Reply:
As already mentioned this is a real censored crash report. I'd like to keep it and maybe add more assertions to avoid regressions.

Reply:
we don't need 700 lines of JSON for a unit test though